### PR TITLE
Reference Articles

### PR DESCRIPTION
**Pull request by albertbori**
## Description

Cleaned up the overall documentation structure and added the following reference articles:

- View Navigation
- View Communication
- View State Extensions

## Type of Change

- [ ] Bug Fix
- [ ] New Feature
- [ ] Breaking Change
- [ ] Refactor
- [x] Documentation
- [ ] Other (please describe)

### CODE REVIEW

File: Sources/VSM/Documentation.docc/Reference/ViewCommunication.md [hunk 1]
```diff
@@ -1,4 +1,159 @@
...
+            addChild(editProfileViewController)
+            ...
+        }
+    }
+}
+```
+
+## Data Communication
+
+This category encompasses the majority of the feature communication scenarios. Any communication between views that does not belong in <doc:ViewCommunication#View-Communication> or <doc:ViewCommunication#Feature-Communication> falls into the this final category.
```
Comment (granoff) [inline Sources/VSM/Documentation.docc/Reference/ViewCommunication.md:153]:
```suggestion
This category encompasses the majority of the feature communication scenarios. Any communication between views that does not belong in <doc:ViewCommunication#View-Communication> or <doc:ViewCommunication#Feature-Communication> falls into this final category.
```

File: Sources/VSM/Documentation.docc/Reference/ViewStateExtensions.md [hunk 2]
```diff
@@ -1,12 +1,232 @@
...
+The SwiftUI framework keeps track of a view's identity in one of two ways, depending on how your code is written:
+
+1. Structural Identity
+1. Explicit Identity
+
+We will cover only Structural Identity in this article, as Explicit Identity is more easily implemented without any VSM considerations
+
+### Structural Identity
+
+Structural Identity in SwiftUI is the default mechanism for identifying views. It is done by recording the location of your view in the view hierarchy. (If you are familiar with identifying HTML tags on a web page in css or javascript by using [XPath](https://www.w3schools.com/xml/xpath_syntax.asp), this is the a very similar concept.)
```
Comment (granoff) [inline Sources/VSM/Documentation.docc/Reference/ViewStateExtensions.md:28]:
```suggestion
Structural Identity in SwiftUI is the default mechanism for identifying views. It is done by recording the location of your view in the view hierarchy. (If you are familiar with identifying HTML tags on a web page in CSS or JavaScript by using [XPath](https://www.w3schools.com/xml/xpath_syntax.asp), this is a very similar concept.)
```

File: Sources/VSM/Documentation.docc/Reference/ViewStateExtensions.md [hunk 3]
```diff
@@ -1,12 +1,232 @@
...
+When working with the SwiftUI framework, you will often come across views or view modifiers that require a `Binding<T>` of some kind. For example, you may have an error view that you want to display if the user gets into an error state. Without view state extensions, your view will be cluttered with code that translates the view state enum into a `Binding<Bool>` and a confusing `.sheet()` view modifier implementation.
+
+```swift
+struct EditUserProfileView: View, ViewStateRendering {
+    ...
+    var hasError: Binding<Bool>
+
+    init() {
+        _hasError = Binding<Bool> {
+            get: {
```
Comment (granoff) [inline Sources/VSM/Documentation.docc/Reference/ViewStateExtensions.md:113]: Is the trailing colon correct here (and for set, below)?

**Review by granoff — APPROVED** [review]

File: Sources/VSM/Documentation.docc/Reference/ViewStateExtensions.md (hunk 3 above)
Comment (albertbori) [reply Sources/VSM/Documentation.docc/Reference/ViewStateExtensions.md:113]: It is not! good catch!

Comment (albertbori) [inline Sources/VSM/Documentation.docc/Reference/ViewStateExtensions.md:113]:
```suggestion
            get {
```

File: Sources/VSM/Documentation.docc/Reference/ViewStateExtensions.md [hunk 4]
```diff
@@ -1,12 +1,232 @@
...
+                if case .savingError(let errorModel) = state.editingState {
+                    return true
+                }
+                return false
+            },
+            set: {
```
Comment (albertbori) [inline Sources/VSM/Documentation.docc/Reference/ViewStateExtensions.md:119]:
```suggestion
            set {
```

**Pull request merged**